validphys2/src/validphys/commondataparser.py

```diff
@@ -89,3 +92,87 @@ def parse_systypes(systypefile):
     systypetable.set_index("sys_index", inplace=True)
 
     return systypetable
+
+def write_commondata(commondata_list, filter_path):
+    """
+    GENERAL DESCRIPTION:
+
+    writes dataset data in filter folder using the commondata file format
+
+    Parameters
+    ----------
+
+    commondata_list : list
+                     commondata object (cuts should be already applied)
+
+    path : str
+         path to filter folder
+
+    
+    """
+
+    for commondata_instance in commondata_list:
+        # path
+        path = str(filter_path) + f'/{commondata_instance.setname}'
+        path_data = str(path) + f"/DATA_{commondata_instance.setname}.dat"
+        commondata_tab = commondata_instance.commondata_table.reset_index(drop = True) # do not use maskcut index
+        commondata_tab.index += 1 # index starting from 1
+        
+        header = f"{commondata_instance.setname} {commondata_instance.nsys} {commondata_instance.ndata}\n"
+        
+        #==== write DATA =====#
+        with open(path_data, "w+") as f:
+            f.write(header)
+            commondata_tab.to_csv(f, sep="\t", header=None)
+
+
+
+def make_systype_dir(path):
+    """
+    GENERAL DESCRIPTION:
+
+    creates directory named systypes 
+
+    Parameters
+    ----------
+
+    path : str
+         path to systypes filter/dataset_name/systypes folder
+    
+
+    """
+    if path.exists():
+        log.warning(f"systypes folder exists: {path} Overwriting contents")
+    else:
+        path.mkdir(exist_ok=True)
+
+
+def write_systype(commondata_list, filter_path):
+    """
+    GENERAL DESCRIPTION:
+
+    writes systype data in filter folder using the systype file format  
+    
+    Parameters
+    ----------
+
+    commondata_list : list
+                    commondata object (cuts should be already applied)
+
+    filter_path : str
+                 path to filter folder
+
+    
+    """
+
+    for commondata_instance in commondata_list: 
+        path = filter_path / commondata_instance.setname / 'systypes'
+        make_systype_dir(path)
+        path_data = str(path) + f"/SYSTYPES_{commondata_instance.setname}_DEFAULT.dat"
+        systype_tab = commondata_instance.systype_table
+        header = f"{len(systype_tab.index)}\n"
```

Comment:
If this is an instance of `coredata:CommonData` it includes already a property with the `nsys` value
```suggestion
        header = f"{commondata_instance.nsys}\n"
```